dags/springer/parser.py

```diff
@@ -0,0 +1,259 @@
+                "ArticleType",
+                extra_function=lambda x: article_type_mapping[x],
+            ),
+            TextExtractor(
+                "dois", "./Journal/Volume/Issue/Article/ArticleInfo/ArticleDOI"
+            ),
+            CustomExtractor("arxiv_eprints", self._get_arxiv_eprints),
+            CustomExtractor("page_nr", self._get_page_nrs),
+            CustomExtractor("abstract", self._get_abstract),
+            TextExtractor(
+                "title",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleTitle",
+            ),
+            CustomExtractor("authors", self._get_authors),
+            TextExtractor(
+                "collaborations",
+                "./Journal/Volume/Issue/Article/ArticleHeader/AuthorGroup/InstitutionalAuthor/InstitutionalAuthorName",
+                False,
+                extra_function=lambda x: [x],
+            ),
+            TextExtractor(
+                "journal_title",
+                "./Journal/JournalInfo/JournalTitle",
+                extra_function=lambda s: s.lstrip("The "),
+            ),
+            TextExtractor(
+                "journal_issue", "./Journal/Volume/Issue/IssueInfo/IssueIDStart"
+            ),
+            TextExtractor(
+                "journal_volume", "./Journal/Volume/VolumeInfo/VolumeIDStart"
+            ),
+            AttributeExtractor("journal_artid", "./Journal/Volume/Issue/Article", "ID"),
+            TextExtractor(
+                "journal_fpage",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleFirstPage",
+            ),
+            TextExtractor(
+                "journal_lpage",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleLastPage",
+            ),
+            TextExtractor(
+                "journal_year",
+                "./Journal/Volume/Issue/Article/ArticleInfo/*/OnlineDate/Year",
+                extra_function=lambda x: int(x),
+            ),
+            CustomExtractor("date_published", self._get_published_date),
+            TextExtractor(
+                "copyright_holder",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleCopyright/CopyrightHolderName",
+            ),
+            TextExtractor(
+                "copyright_year",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleCopyright/CopyrightYear",
+            ),
+            TextExtractor(
+                "copyright_statement",
+                "./Journal/Volume/Issue/Article/ArticleInfo/ArticleCopyright/copyright-statement",
+                False,
+            ),
+            CustomExtractor("license", self._get_license),
+            TextExtractor(
+                "collections",
+                "./Journal/JournalInfo/JournalTitle",
+                extra_function=lambda x: [x.lstrip("The ")],
+            ),
+        ]
+        super().__init__(extractors)
+
+    def _get_abstract(self, article: ET.Element):
+        paragraph = article.find(
```

Comment:
what's the structure of `Para` could we just get the whole content instead of iterating multiple times?

Reply:
```xml
                     <Para ID="Par1">
                        This paper describes a strategy for a general search used by the ATLAS Collaboration to find potential indications of new physics. Events are classified according to their final state into many event classes. For each event class an automated search algorithm tests whether the data are compatible with the Monte Carlo simulated expectation in several distributions sensitive to the effects of new physics. The significance of a deviation is quantified using pseudo-experiments. A data selection with a significant deviation defines a signal region for a dedicated follow-up analysis with an improved background expectation. The analysis of the data-derived signal regions on a new dataset allows a statistical interpretation without the large look-elsewhere effect. The sensitivity of the approach is discussed using Standard Model processes and benchmark signals of new physics. As an example, results are shown for 3.2 fb
                        <InlineEquation ID="IEq1">
                           <InlineMediaObject>
                              <ImageObject Color="BlackWhite" FileRef="10052_2019_6540_Article_IEq1.gif" Format="GIF" Rendition="HTML" Type="Linedraw" />
                           </InlineMediaObject>
                           <EquationSource Format="TEX">
                              <![CDATA[$$^{-1}$$]]>
                           </EquationSource>
                           <EquationSource Format="MATHML">
                              <![CDATA[
                    ]]>
                              <math xmlns:xlink="http://www.w3.org/1999/xlink">
                                 <msup xmlns:xlink="http://www.w3.org/1999/xlink">
                                    <mrow xmlns:xlink="http://www.w3.org/1999/xlink" />
                                    <mrow xmlns:xlink="http://www.w3.org/1999/xlink">
                                       <mo xmlns:xlink="http://www.w3.org/1999/xlink">-</mo>
                                       <mn xmlns:xlink="http://www.w3.org/1999/xlink">1</mn>
                                    </mrow>
                                 </msup>
                              </math>
                              <![CDATA[
                  ]]>
                           </EquationSource>
                        </InlineEquation>
                        of proton–proton collision data at a centre-of-mass energy of 13 
                        <InlineEquation ID="IEq2">
                           <InlineMediaObject>
                              <ImageObject Color="BlackWhite" FileRef="10052_2019_6540_Article_IEq2.gif" Format="GIF" Rendition="HTML" Type="Linedraw" />
                           </InlineMediaObject>
                           <EquationSource Format="TEX">
                              <![CDATA[$$\text {TeV}$$]]>
                           </EquationSource>
                           <EquationSource Format="MATHML">
                              <![CDATA[
                    ]]>
                              <math xmlns:xlink="http://www.w3.org/1999/xlink">
                                 <mtext xmlns:xlink="http://www.w3.org/1999/xlink">TeV</mtext>
                              </math>
                              <![CDATA[
                  ]]>
                           </EquationSource>
                        </InlineEquation>
                        collected with the ATLAS detector at the LHC in 2015, in which more than 700 event classes and more than
                        <InlineEquation ID="IEq3">
                           <InlineMediaObject>
                              <ImageObject Color="BlackWhite" FileRef="10052_2019_6540_Article_IEq3.gif" Format="GIF" Rendition="HTML" Type="Linedraw" />
                           </InlineMediaObject>
                           <EquationSource Format="TEX">
                              <![CDATA[$$10^5$$]]>
                           </EquationSource>
                           <EquationSource Format="MATHML">
                              <![CDATA[
                    ]]>
                              <math xmlns:xlink="http://www.w3.org/1999/xlink">
                                 <msup xmlns:xlink="http://www.w3.org/1999/xlink">
                                    <mn xmlns:xlink="http://www.w3.org/1999/xlink">10</mn>
                                    <mn xmlns:xlink="http://www.w3.org/1999/xlink">5</mn>
                                 </msup>
                              </math>
                              <![CDATA[
                  ]]>
                           </EquationSource>
                        </InlineEquation>
                        regions have been analysed. No significant deviations are found and consequently no data-derived signal regions for a follow-up analysis have been defined.
                     </Para>
                  ```

Reply:
@drjova There is some inner tags that we don't need. The xml lib that I'm using doesn't handle those cases so that's why I need to do all this filtering